wandb/sdk/launch/agent/__init__.py

```diff
@@ -144,10 +144,12 @@ def run_job(self, job):
         backend_config = dict(SYNCHRONOUS=True, DOCKER_ARGS={}, STORAGE_DIR=None)
         if _is_wandb_local_uri(uri):
             backend_config[PROJECT_DOCKER_ARGS]["network"] = "host"
+
+        self._api.ack_run_queue_item(job["runQueueItemId"], project.run_id)
+        wandb.finish()
```

Comment:
I'm confused as to what this does. I didn't think we were in a run context in this process?

Reply:
You're right. Initially instead of generating the ID i was initing a run, forgot to remove this

Reply:
Ah ok. and why ack the item before the run has started running? it might not make a difference depending on how we implement _backend.run but in theory it makes sense to only ack once we know that a run has successfully started.

Reply:
I agree, acking once we know the run is started is better. I'm not exactly sure how to have the launched process communicate that it has successfully inited the run. Would acking right before the runner calls the final docker command be all right?

Reply:
I think the most "correct" way would be to pass an environment variable to the container with the run queue item ID, and then modify the `wandb.init` codepath to send an ack if that environment variable is specified. But that might be more of a can of worms so I'm ok doing something best-effort in here for now.

Reply:
But definitely leave a TODO to that ^ effect

Reply:
This is actually a great idea. I'll have this ack right before launching for now. Since it is a minimal change, but I'll try that out, I think it shouldn't be too bad

Reply:
Filing a ticket for this: https://wandb.atlassian.net/browse/WB-5853